dashboard/models.py

```diff
@@ -171,8 +203,9 @@ class Course(models.Model):
     GRADING_CHOICES = [('Percent', 'Percent'), ('Point', 'Point'), ]
     show_grade_type = models.CharField(verbose_name="Show Grade Type", max_length=255,
                                          choices=GRADING_CHOICES, default='Percent')
+    last_cron_run = models.DateTimeField(verbose_name="Last time cron ran for this course", null=True, blank=True)
```

Comment:
Couple thoughts I had here, but I'm not sure how important they are.

1) I don't love `last_cron_run` as a variable name. `cron` as a term is related to a specific technology, which aren't even using anymore (though we are still using `DjangoCronJob`). I think I'd prefer something like `data_last_updated_at`. That could still potentially be useful in a world where we don't ETL (though maybe not).

2) In its current implementation, I believe this is just going to set all of the `Course.last_cron_run` values to the same value (which is repetitive). If I expected this to be the same in perpetuity, I'd say this needs to be normalized more. `CronRun` or `DataUpdate` could be a table, and we could FK to the `course`, allowing `null`s as an indicator of "this course (and child data) wasn't updated as part of any cron run yet". If we moved to a workflow where individual courses could be updated alone or as a group (suggested recently by @chrisrrowland), then **maybe** it makes sense to do it this way. Still, if you're ever going to run multiple at the same time and not distinguish between when one course was updated and when the next was, a new table and FK may make sense.

My two cents. Feel free to disagree. To a fault, I tend to prefer database normalization, though I know it can take more time and does not always seem worth it. In fact, I think I elected not to do something similar in `placement-exams` recently.

Reply:
I think maybe 1) is worth considering, but given time constraints and possible refactoring in the future, I'm fine with  ignoring 2) for now.

Reply:
1) Okay, that sounds like a good suggestion. Easier to make it now than change it. I think I'm going to drop the `_at` and just call it `data_last_updated`. 
2) Currently this is the case. The only reason that this column was added it that some on the dev calls say they wanted a field in the database to see when the course was updated. We update all courses together at the same time (except for new courses that haven't yet been run)

Because it would just be a 1:1 table and all of these will have a value, I opted for a column rather than a FK to another table. 

I wrote this method so it could take a QuerySet and operate on a limited set of values rather than all of them. I'm planning on fixing another of your suggestions to only update the locked courses, so it's possible a course added during the run won't get updated. There might also be cases we want to disable courses or if all of the views are disabled don't run the ETL on that course. In that case those won't get updated either. But we don't do that yet.

Reply:
I'm fine with whatever you want to do here. But it wouldn't be 1:1, it'd 1:M from `DataUpdate` to `Course`. Most of the courses would share the same timestamp, so to avoid repeating the data in the database, you just store it once on `DataUpdate`.

Reply:
That really becomes a 1:M with the CronJobLog then we wouldn't need a new table.

Maybe that's most appropriate, but I'd also like to remove this library since we really aren't using it and it hasn't been updated since 2018. I think this might be worth fixing if we ever wrote our own more simplified cronjob, but am reluctant to add a FK to that table.

Reply:
I created #1090 to improve this. I don't want to fix this because I'd like to remove django-cron and fixing this correctly would involve a FK to that table. If we had a simpler table to manage the crons (or even just re-used this table locally without that package) it would be easier.

Reply:
Yeah, I'm in favor of removing `django-cron`. Doesn't seem like it would be too much work to just re-write this as our own class. As I've already indicated, no worries about dealing with this now.